src/client/Microsoft.Identity.Client/Utils/OptionalSemaphoreSlim.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.Utils
+{
+    /// <summary>
+    /// An object that either wraps a SemaphoreSlim for syncronization or ignores syncronization completely and just keeps track of Wait / Release operations.
+    /// </summary>
+    internal class OptionalSemaphoreSlim
```

Comment:
This version of MSAL will not use a semaphore and this class simplifies this.

This should be fine in the "1 app per request" paradigm MIW uses.